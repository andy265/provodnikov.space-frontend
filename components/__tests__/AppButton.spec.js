const { createRenderer } = require('vue-server-renderer')
const { mount, shallow } = require('vue-test-utils')
const AppButton = require('~/components/AppButton')

describe('AppButton.vue', () => {
  it('has same HTML structure', () => {
    let wrapper = shallow(AppButton, {
      slots: {
        default: '<span>some button name</span>'
      }
    })
    let renderer = createRenderer()
    // eslint-disable-next-line
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('can be disabled', () => {
    let wrapper = mount({
      template: '<AppButton :disabled="true"/>',
      components: {
        AppButton
      }
    })
    expect(wrapper.find('button').element.getAttribute('disabled')).toBeTruthy()
  })

  it('emit click event', () => {
    let wrapper = shallow(AppButton)
    wrapper.trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
  })
})
