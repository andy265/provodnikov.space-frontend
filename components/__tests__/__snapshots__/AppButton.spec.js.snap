// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppButton.vue has same HTML structure 1`] = `<button class="app-button ripple"><span>some button name</span></button>`;
